### ISSUE TITLE
Compatibility with client side (browser)

### ISSUE THREAD

**Issue by marcio199226** [perry-mitchell/webdav-client#158]
Hi @perry-mitchell i can't run your code in my angular app , this is an example code:
```
    const client = createClient(
      'http://localhost:8081',
      {
          username: 'admin',
          password: 'admin'
      }
    );
    client.getDirectoryContents('/', { headers: { Accept: 'text/html,application/xhtml+xml,application/xml,text/xml;', } }).then((items) => console.log(items));
```

This give me the right response
```
<?xml version="1.0" encoding="UTF-8"?><D:multistatus xmlns:D="DAV:"><D:response><D:href>//</D:href><D:propstat><D:prop><D:resourcetype><D:collection xmlns:D="DAV:"/></D:resourcetype><D:displayname></D:displayname><D:getlastmodified>Thu, 16 May 2019 12:25:51 GMT</D:getlastmodified><D:supportedlock><D:lockentry xmlns:D="DAV:"><D:lockscope><D:exclusive/></D:lockscope><D:locktype><D:write/></D:locktype></D:lockentry></D:supportedlock></D:prop><D:status>HTTP/1.1 200 OK</D:status></D:propstat></D:response><D:response><D:href>/LICENSE</D:href><D:propstat><D:prop><D:getcontentlength>1097</D:getcontentlength><D:getlastmodified>Thu, 16 May 2019 12:03:26 GMT</D:getlastmodified><D:getcontenttype>text/plain; charset=utf-8</D:getcontenttype><D:getetag>"159f283580185910449"</D:getetag><D:resourcetype></D:resourcetype><D:displayname>LICENSE</D:displayname><D:supportedlock><D:lockentry xmlns:D="DAV:"><D:lockscope><D:exclusive/></D:lockscope><D:locktype><D:write/></D:locktype></D:lockentry></D:supportedlock></D:prop><D:status>HTTP/1.1 200 OK</D:status></D:propstat></D:response></D:multistatus>
```

But i have this error:
```
console.js:35 Error: Uncaught (in promise): TypeError: path.basename is not a function
TypeError: path.basename is not a function
    at propsToStat (dav.js:58)
    at directoryContents.js:77
    at Array.map (<anonymous>)
    at getDirectoryFiles (directoryContents.js:68)
    at directoryContents.js:51
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:391)
    at Object.onInvoke (core.js:17298)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:390)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.run (zone.js:150)
    at zone.js:889
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:423)
    at Object.onInvokeTask (core.js:17289)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:422)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.runTask (zone.js:195)
    at drainMicroTaskQueue (zone.js:601)
    at ZoneTask.push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask [as invoke] (zone.js:502)
    at invokeTask (zone.js:1744)
    at XMLHttpRequest.globalZoneAwareCallback (zone.js:1770)
    at resolvePromise (zone.js:831)
    at resolvePromise (zone.js:788)
    at zone.js:892
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:423)
    at Object.onInvokeTask (core.js:17289)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:422)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.runTask (zone.js:195)
    at drainMicroTaskQueue (zone.js:601)
    at ZoneTask.push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask [as invoke] (zone.js:502)
    at invokeTask (zone.js:1744)
    at XMLHttpRequest.globalZoneAwareCallback (zone.js:1770)
```

Because in browser we didn't bring nodejs packages such as path / process

Can you help me ?

P.S issue comes from this line https://github.com/perry-mitchell/webdav-client/blob/master/source/interface/dav.js#L56

**Comment by perry-mitchell** [comment]
Hello @marcio199226 - You need to bundle the node libraries for it to work. Webpack should do this automatically, for instance. The `path.basename` function is required.

In light of your issue, I've gone ahead and mentioned this in the [readme](https://github.com/perry-mitchell/webdav-client/blob/master/README.md#usage-in-the-browser).

**Comment by perry-mitchell** [comment]
I mean, if you go to the trouble of transpiling the other required dependencies like `xmljs`, it should be no problem to include `path` as well.

**Comment by marcio199226** [comment]
i have already webpack through angular cli, I have no trouble with xmljs but after running my dev server my issue is still present.

Do you know how can I resolve it?

**Comment by perry-mitchell** [comment]
I'm not familiar with the build process in angular sorry. My advice would be to indeed look into bundling the path module that webpack uses for supporting that in the browser - it's webpack's job to include such functionality.

I may indeed look to change the `path` dependency into an external module dependency from npm later, so bundling is perhaps simpler, but this should still be fine to do as it stands currently.

As there's no issue with webdav-client, I'll close this issue.

**Comment by marcio199226** [comment]
> As there's no issue with webdav-client, I'll close this issue.

Sorry but the issue is present if it's not working with webpack out of the box...
Please use something like that https://github.com/webpack/node-libs-browser otherwise I have to fork your repo and fix it

**Comment by perry-mitchell** [comment]
@marcio199226 What do you mean? I use this in servals webpack based projects out of the box without issue.. Maybe the angular setup causes issues here. This is not an issue with this project, it’s how you choose to bundle it.

**Comment by perry-mitchell** [comment]
That being said, I have removed the path dependency and will release a new version tomorrow.

**Comment by perry-mitchell** [comment]
Released as 2.8.0

**Comment by marcio199226** [comment]
Ok, when i wrote that comment in 2.7.0 release the issue was still present
